packages/widgetbook_core/lib/src/navigation_tree/extensions/filter_node_extension.dart

```diff
@@ -0,0 +1,42 @@
+import 'package:widgetbook_core/widgetbook_core.dart';
+
+extension FilterNodes on NavigationTreeNodeData {
+  NavigationTreeNodeData? filterNodeByQuery(String searchQuery) {
```

Comment:
This extension is unnecessary since we have complete control over the NavigationTreeNodeData code. Extensions are great for expanding functionality coming from another package whose source code we cannot influence.

Reply:
Why is this code not part of a Bloc/Provider? If you need this on a node, you can also put it into the `NavigationTreeNodeData` class.

Reply:
But I'd like to keep models clean from logic

Reply:
Moved logic from the model into the bloc